CycloneDX/Services/ProjectAssetsFileService.cs

```diff
@@ -48,27 +47,25 @@ public HashSet<NugetPackage> GetNugetPackages(string projectFilePath, string pro
             if (_fileSystem.File.Exists(projectAssetsFilePath))
             {
                 var assetFileReader = _assetFileReaderFactory();
-                string jsonContent = assetFileReader.ReadAllText(projectAssetsFilePath);
-                JsonDocument assetFileObject = _assetJsonObject.Parse(jsonContent);
-                // get all direct nuget dependencies of the project
-                JsonElement frameworksProperties = assetFileObject.RootElement.GetProperty("project").GetProperty("frameworks");
-
                 var assetsFile = assetFileReader.Read(projectAssetsFilePath);
 
                 foreach (var targetRuntime in assetsFile.Targets)
                 {
                     var runtimePackages = new HashSet<NugetPackage>();
+                    var targetFramework = assetsFile.PackageSpec.GetTargetFramework(targetRuntime.TargetFramework);
+                    var dependencies = targetFramework.Dependencies;
+
                     foreach (var library in targetRuntime.Libraries.Where(lib => lib.Type != "project"))
                     {
+                        var libs = dependencies.FirstOrDefault(ld => ld.Name.Equals(library.Name));
```

Comment:
`FirstOrDefault` will return `null` if the library is not part of the `dependencies` section.